### PR TITLE
Rerunlund

### PR DESCRIPTION
**Pull request by rsholmes**
This branch improves upon the previously submitted "rerun" branch.

Defines a new option RERUN_SELECTED="&lt;run&gt;[,&lt;dir&gt;]". &lt;run&gt; is a run number; &lt;dir&gt; is a path to a directory, defaulting to "./".

This option causes GEMC to read files in the given directory with filenames of the form run&lt;run&gt;&lt;evt&gt;.rndm, in numerical order by event number &lt;evt&gt;, and use them to initialize the random number generator prior to the start of each event, thus rerunning the events. If not further limited by the N option, the run is terminated after the last such event is rerun.

If reading from a LUND generator input file, the lines corresponding to non-rerun events are skipped over so that the rerun events are started with the correct input.

Event numbers shown in output files are set to match those of the original events.

### CODE REVIEW

File: src/MEventAction.cc [hunk 1]
```diff
@@ -179,12 +179,18 @@ MEventAction::~MEventAction()
...
 {
+        G4RunManager *runManager = G4RunManager::GetRunManager();;
 	if(gen_action->isFileOpen() == false) {
-		G4RunManager *runManager = G4RunManager::GetRunManager();;
 		runManager->AbortRun();
 		cout << " No more events in the input file." << endl;
 		return;
 	}
+	
+	MPrimaryGeneratorAction* pga = (MPrimaryGeneratorAction*)(runManager->GetUserPrimaryGeneratorAction());
```
Comment (maureeungaro) [inline src/MEventAction.cc]: Can you comment here on why this is done.

Comment (rsholmes) [reply src/MEventAction.cc]: Not sure what you're asking about. The only change in the code excerpt shown is to define pointer pga, which is used below.

Comment (maureeungaro) [reply src/MEventAction.cc]: Not just this line sorry, I meant the following ones as well: why we need to "rerunEvent()" if it is reRun, was the event number not set correctly?

Comment (rsholmes) [reply src/MEventAction.cc]: These lines:

	if (pga->doneRerun())
	  return;

keep GEMC from trying to write out an extra event after the primary generator action has run through all the events to be rerun. 

These lines:

	if (pga->isRerun())
	  evtN = pga->rerunEvent();

are to assign the same event numbers in the output as were used previously. E.g. if we are reading files run2event37.rndm, run2event94.rndm, and run2event122.rndm, the output is three events with event numbers 37, 94, and 122. Without those lines they would have event numbers 1, 2, and 3.

**Pull request merged**